From: Danny Kukawka <dkukawka@suse.de>
Date: Thu, 5 Jun 2014 00:03:44 +0200
Subject: [PATCH] init-radosgw: adjust for opensuse

Merge distro patches to git.

This was radosgw-init-opensuse.patch

(cherry picked from commit 5cd81b3241061be1caf9fbe0eda34f4a0d3f945a)
---
 src/init-radosgw | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/init-radosgw b/src/init-radosgw
index 325b877..26ebff7 100644
--- a/src/init-radosgw
+++ b/src/init-radosgw
@@ -35,7 +35,7 @@ done
 PREFIX='client.radosgw.'
 
 # user to run radosgw as (it not specified in ceph.conf)
-DEFAULT_USER='www-data'
+DEFAULT_USER='wwwrun'
 
 RADOSGW=`which radosgw`
 if [ ! -x "$RADOSGW" ]; then
@@ -78,7 +78,7 @@ case "$1" in
             fi
 
             echo "Starting $name..."
-            start-stop-daemon --start -u $user -x $RADOSGW -- -n $name
+	     /sbin/startproc -u $user $RADOSGW
         done
         daemon_is_running $RADOSGW
         ;;
@@ -94,7 +94,8 @@ case "$1" in
         start-stop-daemon --stop -x $RADOSGW --oknodo
         ;;
     status)
-        daemon_is_running $RADOSGW
+	 /sbin/checkproc $RADOSGW
+	 rc_status -v
         ;;
     *)
         echo "Usage: $0 {start|stop|restart|force-reload|reload|status} [-v|--verbose]" >&2
